adserver/tasks.py

```diff
@@ -0,0 +1,52 @@
+"""Celery tasks for the ad server."""
+import logging
+
+from django.db.models import Count
+
+from .models import GeoImpression
+from .models import Offer
+from .utils import get_ad_day
+from config.celery_app import app
+
+log = logging.getLogger(__name__)  # noqa
+
+
+@app.task()
+def daily_update_geos():
+    """Update the Geo index each day."""
+    beginning_of_today = get_ad_day()
+    log.info("Updating GeoImpressions for %s", beginning_of_today)
+
+    for values in (
+        Offer.objects.filter(date__gte=beginning_of_today, viewed=True)
```

Comment:
I would also constrain on `lt=beginning_of_day + timedelta(days=1)` or something like that. There could be some timing issues with this code right around midnight UTC.